CMakeLists.txt

```diff
@@ -13,6 +13,9 @@ if(NOT DEFINED SMALL_EMBEDDED)
     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -Werror")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -Werror")
+else()
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
```

Comment:
Why do test/app-tap modules require this? Is it only required for ASAN builds? Please add a proper explanation here.

Reply:
I removed this hunk. I added it as there were compilations errors for `module_api` test library. I guess I mistakenly added `small` as dependency for this library because I needed to add include path from small library.

Instead now I removed this hunk too (part of this PR): 
```diff
-    # In embedded case we should able to find small_config.h.
-    target_include_directories(${PROJECT_NAME} INTERFACE
-                               "${CMAKE_CURRENT_BINARY_DIR}/small/include")
-
```
and added to Tarantool instead:
```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -300,6 +300,7 @@ include_directories(${PROJECT_SOURCE_DIR}/src)
 include_directories(${PROJECT_BINARY_DIR}/src)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib/small/include)
+include_directories(${PROJECT_BINARY_DIR}/src/lib/small/small/include)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib/small/third_party)
 include_directories(${PROJECT_SOURCE_DIR}/src/lib/core)
 include_directories(${PROJECT_SOURCE_DIR}/third_party)
```

So that `module_api` and `on_shutdown` can find `small_config.h`